src/service.py

```diff
@@ -50,11 +70,54 @@ def update_project_details_(project: Projects, project_data: Project, db: Sessio
         project.name = project_data.name
     if project_data.description:
         project.description = project_data.description
-
     db.commit()
     db.refresh(project)
 
 
 def delete_project_(project: Projects, db: Session) -> None:
+    db.query(UserProject).filter(UserProject.project_id == project.id).delete()
     db.delete(project)
     db.commit()
+
+
+def create_user_(db: Session, create_user_request: User) -> None:
+    h = hashlib.new("SHA256")
+    h.update(create_user_request.password.encode())
+    password_hash = h.hexdigest()
+    create_user_model = Users(
+        id=uuid.uuid4(),
+        name=create_user_request.name,
+        email=create_user_request.email,
+        hashed_password=password_hash,
+    )
+    db.add(create_user_model)
+    db.commit()
+
+
+def authenticate_user(username: str, password: str, db: Session) -> Users | None:
```

Comment:
this function should take a user object. YOu do two things here. utilize existing function to get a user